lms/services/lis_result_sourcedid.py

```diff
@@ -8,6 +8,40 @@ class LISResultSourcedIdService:
 
     def __init__(self, _context, request):
         self._db = request.db
+        self._authority = request.registry.settings["h_authority"]
+
+    def fetch_students_by_assignment(
+        self, oauth_consumer_key, context_id, resource_link_id
+    ):
+        """
+        Fetch data for students having LIS result records for an assignment.
+
+        Retrieve all :class:`~lms.models.LISResultSourcedId`s that match this
+        assignment (each unique combination of (``oauth_consumer_key``,
+        ``context_id``, ``resource_link_id``) corresponds to an assignment).
+        There should be one record per applicable student who has launched this
+        assignment (and had ``LISresultSourcedId`` data persisted for them).
+
+        :arg oauth_consumer_key: Which LMS application install the request
+                                 corresponds to.
+        :type oauth_consumer_key: str
+        :arg context_id: LTI parameter indicating the course
+        :type context_id: str
+        :arg resource_link_id: LTI parameter (roughly) equating to an assignment
+        :type resource_link_id: str
+        :rtype: list[:class:`lms.models.LISResultSourcedId`]
+        """
+        lis_result_sourcedids = (
+            self._db.query(LISResultSourcedId)
+            .filter_by(
+                oauth_consumer_key=oauth_consumer_key,
+                context_id=context_id,
+                resource_link_id=resource_link_id,
+            )
+            .all()
+        )
+
+        return lis_result_sourcedids
```

Comment:
I think `return self._db.query(LISResultSourcedId).filter_by(...)` should work here: sqlalchemy query objects are already iterable, there's no need to turn them into a list by calling `all()`

Reply:
Let me know how passionate you are about this `.all` change— I am uncertain as to whether I'd need to rewrite the mocking in the `frontend_app` tests ( https://github.com/hypothesis/lms/pull/981/files#diff-5d331d455031abd0501bd0a0d79b8b49R77 ) to be "safe."